### PR TITLE
Npm@8

### LINKED ISSUE #3106: Migrate to `npm@8`
### What should we do?

As decided we migrate our code base, including this app [to `npm@8`](https://github.com/bpmn-io/internal-docs/issues/482).

### Why should we do it?

Ensure we have a smooth transition.

---

Requires https://github.com/bpmn-io/dmn-js-properties-panel/pull/46 to be merged, released, and integrated.

### PR DESCRIPTION
**Pull request by barmac**
Closes #3106

### CODE REVIEW

**Comment by barmac** [comment]
In npm@8 monorepo, if you run npm install --production i in a subpackage, it is actually run also for the root. Thus, it removes the devdeps from the root. Electron-builder runs this command at build time, so effectively it removes itself from the root, and makes it impossible to build the application with npm@8 until I find a solution.

**Comment by barmac** [comment]
Of course I still need to find out why some tests suddenly started to fail.

**Comment by nikku** [comment]
> https://github.com/camunda/camunda-modeler/pull/3108#issuecomment-1228403180

Maybe [an upgrade helps here](https://github.com/camunda/camunda-modeler/pull/3057)?

**Comment by barmac** [comment]
I will try this out, thanks for the idea!

**Comment by barmac** [comment]
It still fails, unfortunately.

**Comment by barmac** [comment]
OK so now the app builds both for dev and production. I identified two issues which need to be fixed yet:

1. The `app-plugins` protocol stopped to work, as the test plugin logo fails to load with _Failed to load resource: net::ERR_UNSAFE_REDIRECT_
2. There are 14 test cases in several places of the client test suite which started to fail.

**Comment by barmac** [comment]
> There are 14 test cases in several places of the client test suite which started to fail.
I was able to fix this locally. Let's see if it works on CI. I am not sure what exactly caused the issue but it disappeared when I reinstalled modules and removed electron-builder update.

**Comment by barmac** [comment]
> The `app-plugins` protocol stopped to work, as the test plugin logo fails to load with _Failed to load resource: net::ERR_UNSAFE_REDIRECT_

I was able to fix this issue with Electron downgrade from 19.0.14 to 19.0.13. To ensure we don't lose it, I will pin the framework.

**Comment by barmac** [comment]
As each of the issues I found are resolved, this PR is ready for review.

File: lerna.json [hunk 1]
```diff
@@ -1,7 +1,9 @@
 {
+  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
   "version": "5.2.0",
   "packages": [
```
Comment (nikku) [inline lerna.json:4]: I believe this is no longer necessary.

File: package.json [hunk 2]
```diff
@@ -7,12 +7,12 @@
     "app:auto-test": "npm run app:test -- --watch",
     "app:dev": "electron ./dev.js --force-update-checks resources/diagram/simple.bpmn",
     "app:test": "nyc --reporter lcov --exclude \"app/test/**\" --exclude \"app/**/__tests__\" mocha --require ./app/test/expect \"app/**/*-spec.js\"",
-    "client:build": "(cd client && npm run build)",
-    "client:auto-test": "(cd client && npm run auto-test)",
-    "client:dev": "(cd client && npm run dev)",
-    "client:test": "(cd client && npm run test)",
-    "bpmn-io-modelers:test": "(cd client && npm run bpmn-io-modelers:test)",
-    "bpmn-io-modelers:auto-test": "(cd client && npm run bpmn-io-modelers:auto-test)",
+    "client:build": "npm run build --workspace=client",
```
Comment (nikku) [inline package.json:10]: :+1:

**Review by nikku — CHANGES_REQUESTED** [review]
Good job. Just one remark. Already looks great overall! :)

File: lerna.json (hunk 1 above)
Comment (barmac) [reply lerna.json]: Done via https://github.com/camunda/camunda-modeler/pull/3108/commits/44a9de831761252c4a5b86062f8e2cb2d02ebde4

**Review by nikku — APPROVED** [review]

**Pull request merged**